infra/RepoManager.py

```diff
@@ -0,0 +1,209 @@
+
+    Returns:
+      True if the current repo_dir is a valid git repo
+    """
+    git_path = os.path.join(self.repo_dir, '.git')
+    return os.path.isdir(git_path)
+
+  def _commit_exists(self, commit):
+    """ Checks to see if a commit exists in the project repo.
+
+    Args:
+      commit: The commit SHA you are checking for
+      project_name: The name of the project you are checking
+
+    Returns:
+      True if the commit exits in the project
+    """
+
+    # Handle the default case
+    if commit.rstrip() == '':
+      return False
+
+    out, _ = self._run_command(['git', 'branch', '--contains', commit],
+                               self.repo_dir)
+    if ('error: no such commit' in out) or (
+        'error: malformed object name' in out) or (out == ''):
+      return False
+    return True
+
+  def get_current_commit(self):
+    """Gets the current commit SHA of the repo.
+
+    Returns:
+      The current active commit SHA
+    """
+    out, err = self._run_command(['git', 'rev-parse', 'HEAD'], self.repo_dir)
+    if err is not None:
+      return 1
+    return out.strip('\n')
+
+  def get_commit_list(self, old_commit, new_commit):
+    """Gets the list of commits(inclusive) between the old and new commits.
+
+    Args:
+      old_commit: The oldest commit to be in the list
+      new_commit: The newest commit to be in the list
+
+    Returns:
+      The list of commit SHAs from newest to oldest
+
+    Raises:
+      RepoManagerError when commits dont exist
+    """
+
+    if not self._commit_exists(old_commit):
+      raise RepoManagerError(
+          'The old commit %s does not exist' % old_commit)
+    if not self._commit_exists(new_commit):
+      raise RepoManagerError(
+          'The new commit %s does not exist' % new_commit)
+    if old_commit == new_commit:
+      return [old_commit]
+    out, err = self._run_command(
+        ['git', 'rev-list', old_commit + '..' + new_commit], self.repo_dir)
+    result = out.split('\n')
+    result = [i for i in result if i]
+    if err is not None or result == []:
+      raise RepoManagerError('Error gettign commit list between %s and %s '
+                                 % (old_commit, new_commit))
+
+    # Make sure result is inclusive
+    result.append(old_commit)
+    return result
+
+  def checkout_commit(self, commit):
+    """Checks out a specific commit from the repo.
+
+    Args:
+      commit: The commit SHA to be checked out
+
+    Raises:
+      RepoManagerError when checkout is not successful
+    """
+    if not self._commit_exists(commit):
+      print('Commit %s does not exist in current branch' % commit)
+      raise RepoManagerError(
+          'Commit %s does not exist in current branch' % commit)
+
+    git_path = os.path.join(self.repo_dir, '.git', 'shallow')
+    if os.path.exists(git_path):
+      _, err = self._run_command(['git', 'fetch', '--unshallow'], self.repo_dir)
+      if err is not None:
```

Comment:
Rather than checking if `_run_command` failed everywhere and then raising an exception if it does, I would make it raise an exception if an optional argument `check` is `True` and the command fails. Like Python's `subprocess.run`
This will be cleaner.